src/Order/OrderAPI/Controllers/CartsController.cs

```diff
@@ -0,0 +1,128 @@
+using API.Abstraction.Helpers;
+using DAL.Orders;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using OrderAPI.DTOs;
+using OrderApplication.Entities;
+using Carts = OrderApplication.Carts;
+
+namespace OrderAPI.Controllers
+{
+    [ApiController]
+    [Route("orders/carts")]
+    public class CartsController : ControllerBase
+    {
+        private readonly IMediator _mediator;
+        private readonly LinkGenerator _linkGenerator;
+
+        public CartsController(IMediator mediator, LinkGenerator linkGenerator)
+        {
+            _mediator = mediator;
+            _linkGenerator = linkGenerator;
+        }
+
+        [HttpGet("{id:guid}")]
+        [ProducesResponseType(typeof(Resource<IList<CartItem>>), 200)]
+        [ProducesResponseType(404)]
+        public async Task<IActionResult> GetDetails(Guid id)
+        {
+            var details = await _mediator.Send(new Carts.Details.Query { Id = id });
+            if (details is null) return NotFound();
+            var resource = GetResource(details, id);
+            return Ok(resource);
+        }
+
+        [HttpPut("{id:guid}/book")]
+        [ProducesResponseType(typeof(Resource<long>), 200)]
+        [ProducesResponseType(404)]
+        [ProducesResponseType(403)]
+        public async Task<IActionResult> BookCartItems(Guid id)
+        {
+            var result = await _mediator.Send(new Carts.Book.Command { Id = id });
+            if (result is null) return NotFound($"Cart with id '{id}' doesn't exist.");
+            if (result.Error) return BadRequest(result.ErrorMessage);
+            
```

Comment:
I see you have `[ProducesResponseType(403)]`, but there is no code, which tends to return this status code. However - you return BadRequest from code, but don't have documentation for it.

Reply:
Sure, I need to fix that.